src/components/custom-constellation/field/HMRC_ODX_GDSTextPresentation/StatusWork.jsx

```diff
@@ -0,0 +1,26 @@
+import { Status } from "@pega/cosmos-react-core";
+
+export default function StatusWorkRenderer({ value }) {
+  let variant = "info";
+
+  const warnStrings = ["fail", "cancel", "reject", "revoke", "stopped", "warn"];
+  const infoStrings = ["open", "hold", "info", "new"];
+  const successStrings = ["resolved", "completed", "success"];
+  const pendingStrings = ["pending"];
+
+  if (new RegExp(warnStrings.join("|"), "i").test(value)) {
+    variant = "warn";
+  } else if (new RegExp(infoStrings.join("|"), "i").test(value)) {
+    variant = "info";
+  } else if (new RegExp(successStrings.join("|"), "i").test(value)) {
+    variant = "success";
+  } else if (new RegExp(pendingStrings.join("|"), "i").test(value)) {
+    variant = "pending";
+  }
```

Comment:
This should be move to utils. Also i don't see point to use `RegExp` here as we have array methods available to do the same job. if im not missing something :)